src/pods/user-settings/user-settings.component.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+  Typography,
+  Checkbox,
+  FormControlLabel,
+  Button,
+  TextField,
+  Container,
+} from '@material-ui/core';
+import { SettingsEntity } from 'core/settings';
+import produce from 'immer';
+
+interface Props {
+  onSave: (settings: SettingsEntity) => void;
+  onCancel: () => void;
+  userSettings: SettingsEntity;
+}
+
+export const UserSettingsComponent: React.FC<Props> = props => {
+  const { onSave, onCancel, userSettings } = props;
+  const [temporalSettings, setTemporalSettings] = React.useState<
```

Comment:
Maybe is an overkill, but I think is a good practice to introduce here Formik validation.

By using Formik you don't need to create a _temporalSettings_ 

You don't need to add right now Fonk, because you have added a control to avoid validation (an slider I guess)